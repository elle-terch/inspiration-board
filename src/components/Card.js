import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {

  handleClick = () => {
    console.log(`button was clicked! ID = ${this.props.card.id}`)
    this.props.deleteCardCallback(this.props.card.id)
  }

  render(props) {
    console.log(this.props.card.id)

    let icon = this.props.card.emoji;
    if (icon !== null) {
      icon = emoji.getUnicode(icon);
    }


    return (

      <div className="card">
        <div className="card__content">
          <p className="card__content-text">
            {this.props.card.text}
          </p>

          <p className="card__content-emoji">
            {icon}
          </p>
          <button onClick={this.handleClick} type="submit">Delete</button>


        </div>
      </div>
    )
  }
}

Card.propTypes = {
  emoji: PropTypes.string,
  text: PropTypes.string
};

export default Card;
