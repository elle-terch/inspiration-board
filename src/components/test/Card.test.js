import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';
import emoji from 'emoji-dictionary';


const CARD = {
  card:
  {
    emoji: ":)",
    id: 213,
    text: "test text"
  }
}

describe('Card', () => {
  test('that it matches an existing snapshot', () => {
    // First Mount the Component in the testing DOM
    // Arrange
    const wrapper = shallow( <Card
      emoji = {CARD.emoji}
      text = {CARD.text}
      />);

    // Assert that it looks like the last snapshot
    expect(wrapper).toMatchSnapshot();
  });
});
