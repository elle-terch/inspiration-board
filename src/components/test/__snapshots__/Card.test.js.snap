// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card that it matches an existing snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card__content"
  >
    <p
      className="card__content-text"
    />
    <p
      className="card__content-emoji"
    />
    <button
      type="submit"
    >
      Delete
    </button>
  </div>
</div>
`;
